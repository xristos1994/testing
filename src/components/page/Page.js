import React from "react";
import Counter from "./../counter/";

export const Page = () => {
  return (
    <div>
      <h1 data-testid="pageTitle">TEST PAGE</h1>
      <Counter />
    </div>
  );
};

export default Page;
