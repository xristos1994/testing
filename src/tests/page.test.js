import React from "react";
import { render, cleanup } from "@testing-library/react";
import Page from "./../components/page";

afterEach(cleanup);

test("<Page>", () => {
  const { getByTestId, queryByTestId, container } = render(<Page />);
  expect(getByTestId("pageTitle").textContent).toBe("TEST PAGE");
  expect(queryByTestId("value")).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
