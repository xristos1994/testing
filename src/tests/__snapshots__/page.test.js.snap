// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page> 1`] = `
<div>
  <h1
    data-testid="pageTitle"
  >
    TEST PAGE
  </h1>
  <div>
    <button
      data-testid="removeBtn"
    >
      Reduce
    </button>
    <button
      data-testid="addBtn"
    >
      Add
    </button>
    <h1
      data-testid="valueHeader"
    >
      The value is:
    </h1>
    <p
      data-testid="value"
    >
      0
    </p>
  </div>
</div>
`;
